File: app/schemas/goal.py
from pydantic import BaseModel, ConfigDict
from datetime import datetime
from typing import Optional
from enum import Enum



class GoalStatus(str, Enum):
    completed = "completed"
    active = "active"
    abandoned = "abandoned"

class GoalCreate(BaseModel):
    goal_name: str
    description: Optional[str] = None
    priority: int
    start_date: datetime
    end_date: Optional[datetime] = None
    status: GoalStatus

class GoalUpdate(BaseModel):
    goal_name: str | None = None
    description: str | None = None
    priority: int | None = None
    start_date: datetime | None = None
    end_date: datetime | None = None
    status: GoalStatus | None = None

class GoalRead(BaseModel):
    goal_id: int
    goal_name: str
    description: Optional[str] = None
    priority: int
    start_date: datetime
    end_date: Optional[datetime] = None
    created_at: datetime
    status: GoalStatus
    user_id: int

    model_config = ConfigDict(from_attributes=True)

File: web-app/src/types/goal.ts
export type GoalStatus = "active" | "completed" | "abandoned";
export type GoalPriority = 1 | 2 | 3;

export interface GoalCreate {
  goal_name: string;
  description?: string | null;
  priority: GoalPriority;
  start_date: string;
  end_date?: string | null;
  status: GoalStatus;
}

export interface GoalUpdate {
  goal_name?: string | null;
  description?: string | null;
  priority?: GoalPriority | null;
  start_date?: string | null;
  end_date?: string | null;
  status?: GoalStatus | null;
}

export interface GoalRead {
  goal_id: number;
  goal_name: string;
  description: string | null;
  priority: GoalPriority;
  start_date: string;
  end_date: string | null;
  created_at: string;
  status: GoalStatus;
  user_id: number;
}
